src/checkers/inference/InferrableAnnotatedTypeFactory.java

```diff
@@ -0,0 +1,22 @@
+package checkers.inference;
+
+import org.checkerframework.framework.type.AnnotatedTypeFactory;
+import org.checkerframework.framework.type.treeannotator.TreeAnnotator;
+
+/**
+ * Interface for all annotatedTypeFactory that wish to be used with
```

Comment:
It's not for all ATFs that are usable from CFI... just ones that want particular adaptations.